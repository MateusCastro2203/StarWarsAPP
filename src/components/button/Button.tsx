import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {styles} from './styles';
interface ButtonProps {
  isFavorite: boolean;
  onPress: () => void;
}

const Button: React.FC<ButtonProps> = ({isFavorite, onPress}) => {
  return (
    <TouchableOpacity
      onPress={onPress}
      style={[
        styles.button,
        isFavorite ? styles.favoriteButton : styles.addButton,
      ]}>
      <Text style={styles.text}>
        {isFavorite ? 'Remove from favorites' : 'Add to favorites'}
      </Text>
    </TouchableOpacity>
  );
};

export default Button;
