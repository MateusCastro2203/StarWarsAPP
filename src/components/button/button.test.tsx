import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import Button from './Button'; // Ajuste o caminho de importação conforme necessário

describe('Button Component', () => {
  const onPressMock = jest.fn();

  test('snapshot', () => {
    const tree = render(
      <Button isFavorite={false} onPress={onPressMock} />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('deve exibir "Add to favorites" quando isFavorite é false', () => {
    render(<Button isFavorite={false} onPress={onPressMock} />);
    expect(screen.getByText('Add to favorites')).toBeTruthy();
  });

  test('deve exibir "Remove from favorites" quando isFavorite é true', () => {
    render(<Button isFavorite={true} onPress={onPressMock} />);
    expect(screen.getByText('Remove from favorites')).toBeTruthy();
  });
  test('deve chamar onPress quando o botão é clicado', () => {
    render(<Button isFavorite={false} onPress={onPressMock} />);

    fireEvent.press(screen.getByText('Add to favorites'));
    expect(onPressMock).toHaveBeenCalled();
  });
});
