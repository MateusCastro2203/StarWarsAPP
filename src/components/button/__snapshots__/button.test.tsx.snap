// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "green",
      "borderRadius": 5,
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Add to favorites
  </Text>
</View>
`;
